test/src/materials/ImplicitStateful.C

```diff
@@ -0,0 +1,53 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+#include "ImplicitStateful.h"
+
+template<>
+InputParameters validParams<ImplicitStateful>()
+{
+  InputParameters params = validParams<Material>();
+  params.addRequiredParam<std::string>("prop_name", "bla");
+  params.addRequiredParam<std::string>("coupled_prop_name", "bla");
+  params.addRequiredParam<bool>("add_time", "bla");
+  params.addRequiredParam<bool>("older", "bla");
+  return params;
+}
+
+ImplicitStateful::ImplicitStateful(const InputParameters & pars) :
```

Comment:
Actually the get member function is older (pre-dates the getParam wrapper). The wrapper just called `_pars.get` which is a reference to the `InputParameters` object that's passed in. This is one of those "that ship has sailed" arguments. I always preferred using the local parameter objects better but I was overruled in a vote long ago so the wrapper is the preferred mechanism. Please use the wrapper here.